.github/workflows/daily-builds.yml

```diff
@@ -6,12 +6,40 @@ on:
   workflow_dispatch:
 
 jobs:
+  check_history:
+    runs-on: [self-hosted, linux, large]
+    name: Check for new commits
+    outputs:
+      should_run: ${{ steps.check_log.outputs.should_run }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Check for new commits
+        id: check_log
+        run: |
+          git rev-list --abbrev-commit --pretty=oneline HEAD --not $(git rev-list -n1 --before="24 hours" --first-parent HEAD) -- checkbox-ng checkbox-support providers checkbox-core-snap checkbox-snap
+          changes=$(git rev-list --abbrev-commit --pretty=oneline HEAD --not $(git rev-list -n1 --before="24 hours" --first-parent HEAD) -- checkbox-ng checkbox-support providers checkbox-core-snap checkbox-snap)
+          if [ -z $changes ]
```

Comment:
This whole block is a mess. Doing this logic here is unnecessary, all shell logic branches are simply dangerous. This was there before but because this codes gets improved I think it's time to fix it.

Compare all of that big block with:
```
git rev-list --abbrev-commit --pretty=oneline HEAD --not $(git rev-list -n1 --before="24 hours" --first-parent HEAD) -- checkbox-ng checkbox-support providers checkbox-core-snap checkbox-snap >> $GITHUB_OUTPUT
cat The differences: \n $GITHUB_OUTPUT
```



That's it. Linear flow, no branches, less code, same functionality, zero repetition (instead of 3x).

Then, the consumer has only to check if the file is empty.